R/teal_slice.R

```diff
@@ -477,50 +511,52 @@ c.teal_slices <- function(...) {
   )
 }
 
+#' @rdname teal_slice
+#' @keywords internal
+#' @export
+#'
+as.list.teal_slices <- function(x, ...) {
+  slices_list <- lapply(unclass(x), as.list)
+  attrs <- attributes(unclass(x))
+  tss_list <- list(slices = slices_list, attributes = attrs)
+  Filter(Negate(is.null), tss_list) # drop attributes if empty
+}
 
-# format method for teal_slices
+#' @param show_all `logical(1)` should parameters set to NULL be returned
+#' @param nchar `integer(1)` or `NULL` indicating the number of character of the output width
 #' @export
 #' @rdname teal_slice
 #' @keywords internal
 #'
-format.teal_slices <- function(x, show_all = FALSE, ...) {
-  res <- character(0)
-  for (i in seq_along(x)) {
-    ind <- names(x)[i]
-    if (is.null(ind)) ind <- sprintf("[[%d]]", i)
-    res <- append(res, ind)
-    res <- append(res, format(x[[i]], show_all = show_all, ...))
-  }
-
-  includes <- attr(x, "include_varnames")
-  if (length(includes) > 0L) {
-    res <- append(res, "\nfilterable variables:")
-    for (i in seq_along(includes)) {
-      res <- append(res, sprintf(" $ %s: %s", names(includes)[i], toString(includes[[i]])))
+format.teal_slices <- function(x, show_all = FALSE, nchar = 40, ...) {
```

Comment:
`show_all` is not used at all. NULL fields are always printed.

Reply:
yeah, valid point - fixed here 4fcf71d641eba61e5038bad0faeb89863a2d979d